src/Backend/Utils/PlanetCreator.ts

```diff
@@ -0,0 +1,214 @@
+      y: planet.y,
+    };
+
+    const perlinValue = perlin(planetCoords, {
+      key: initializers.SPACETYPE_KEY,
+      scale: initializers.PERLIN_LENGTH_SCALE,
+      mirrorX: initializers.PERLIN_MIRROR_X,
+      mirrorY: initializers.PERLIN_MIRROR_Y,
+      floor: true,
+    });
+
+    const biomeBase = perlin(planetCoords, {
+      key: initializers.BIOMEBASE_KEY,
+      scale: initializers.PERLIN_LENGTH_SCALE,
+      mirrorX: initializers.PERLIN_MIRROR_X,
+      mirrorY: initializers.PERLIN_MIRROR_Y,
+      floor: true,
+    });
+
+    return {
+      location: location,
+      planetCoords: planetCoords,
+      perlinValue: perlinValue,
+      biomeBase: biomeBase,
+    };
+  }
+
+  async createPlanet(planet: AdminPlanet, initializers: LobbyInitializers) {
+    const planetData = this.generatePlanetData(planet, initializers);
+    try {
+      const locNum = planetData.location as BigNumberish;
+
+      const args = Promise.resolve([
+        {
+          location: locNum,
+          perlin: planetData.perlinValue,
+          level: planet.level,
+          planetType: planet.planetType,
+          requireValidLocationId: planet.requireValidLocationId,
+          isTargetPlanet: planet.isTargetPlanet,
+          isSpawnPlanet: planet.isSpawnPlanet,
+        },
+      ]);
+
+      const txIntent: UnconfirmedCreateArenaPlanet = {
+        methodName: ContractMethodName.CREATE_ARENA_PLANET,
+        contract: this.contract.contract,
+        args: args,
+      };
+
+      const tx = await this.contract.submitTransaction(txIntent, {
+        gasLimit: '15000000',
+      });
+
+      await tx.confirmedPromise;
+    } catch (e) {}
+  }
+
+  async revealPlanet(
+    planet: AdminPlanet,
+    initializers: LobbyInitializers,
+  ) {
+    const planetData = this.generatePlanetData(planet, initializers);
+    const getArgs = async () => {
+      const revealArgs = await this.makeRevealProof(
+        planet.x,
+        planet.y,
+        initializers.PLANETHASH_KEY,
+        initializers.SPACETYPE_KEY,
+        initializers.PERLIN_LENGTH_SCALE,
+        initializers.PERLIN_MIRROR_X,
+        initializers.PERLIN_MIRROR_Y,
+        initializers.DISABLE_ZK_CHECKS,
+        initializers.PLANET_RARITY
+      );
+      return revealArgs;
+    };
+
+    const worldLocation = {
+      coords: planetData.planetCoords as WorldCoords,
+      hash: location.toString() as LocationId,
+      perlin: planetData.perlinValue,
+      biomebase: planetData.biomeBase,
+    } as WorldLocation;
+
+    const txIntent: UnconfirmedReveal = {
+      methodName: ContractMethodName.REVEAL_LOCATION,
+      contract: this.contract.contract,
+      locationId: location.toString() as LocationId,
+      location: worldLocation,
+      args: getArgs(),
+    };
+
+    // Always await the submitTransaction so we can catch rejections
+    const tx = await this.contract.submitTransaction(txIntent);
+    console.log(`reveal tx submitted`);
+
+    await tx.confirmedPromise;
```

Comment:
if you say `txReceipt = await tx.confirmedPromise` you can access the receipt data